include/ErrorAnnotation.h

```diff
@@ -0,0 +1,15 @@
+#ifndef ERROR_ANNOTATION_H
+#define ERROR_ANNOTATION_H
+
+// this is not an annotation for an error but effectively takes the place of an
+// error message used when we are checking if a raw annotation string is correct
+// and turns out it is not
```

Comment:
When you write a comment like this one that includes something that a thing is and something that the thing is not, it's usually best to lead with the thing that it is. In this case, that means putting everything after the "but" in this comment first, and then adding as a note that this does not represent an error in the run-time sense.

Reply:
The reason for this advice is that someone reading this file for the first time doesn't necessarily know the whole context. Instead, they are probably looking for an explanation for what this file **is** as opposed to what it **is not**. And we'd prefer to optimize comments for someone who is confused, since they're supposed to be explanatory :)